pkg/healthautoexport/types.go

```diff
@@ -99,12 +114,21 @@ func (m *Metric) UnmarshalJSON(bytes []byte) error {
 		if err := jsoniter.Unmarshal(intermediate.Data, &sa); err != nil {
 			return err
 		}
-		// only process as SleepAnalysis if Value field is set
-		if len(sa) > 0 && sa[0].Value != "" {
+		// only process as SleepAnalysis if first item parses to non-empty SleepAnalysis
+		if *sa[0] != (SleepAnalysis{}) {
 			m.SleepAnalyses = sa
 			return nil
 		}
-		// process aggregated sleep_analysis like other fields
+		var agg []*AggregateSleepAnalysis
+		if err := jsoniter.Unmarshal(intermediate.Data, &agg); err != nil {
+			return err
+		}
+		// only process as AggregatedSleepAnalysis if first item parses to non-empty AggregatedSleepAnalysis
+		if *agg[0] != (AggregateSleepAnalysis{}) {
```

Comment:
Do we need to check the length of `agg`?

Reply:
Let's do so.